Add warning on config in promise usage (#2890)
webpack/webpack.js.org: src/content/configuration/configuration-types.md

};
```

> W> Keep in mind this only works when using CLI. [`webpack()`](/api/node/#webpack) expects a config object.

Reviewer: ```suggestion
W> Returning a `Promise` only works when using webpack via CLI. [`webpack()`](/api/node/#webpack) expects an object.
```
